lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -160,6 +463,94 @@ export default function getAPIImplementation(): APIInterface {
                 userContext,
             });
 
+            if (response.status === "OK") {
+                let userId = response.userId;
+                let email = response.email;
+                async function verifyUser(rUserId: string) {
+                    const emailVerificationInstance = EmailVerification.getInstance();
+                    if (emailVerificationInstance) {
+                        const tokenResponse = await emailVerificationInstance.recipeInterfaceImpl.createEmailVerificationToken(
+                            {
+                                userId: rUserId,
+                                email,
+                                userContext,
+                            }
+                        );
+
+                        if (tokenResponse.status === "OK") {
+                            await emailVerificationInstance.recipeInterfaceImpl.verifyEmailUsingToken({
+                                token: tokenResponse.token,
+                                userContext,
+                            });
+                        }
+                    }
+                }
+                let user = await getUser(userId);
+                if (user === undefined) {
+                    throw Error("this error should never be thrown");
+                }
+                /**
+                 * check if the user is a primaryUser
+                 */
+                if (user.isPrimaryUser) {
+                    /**
+                     * check if there exists an emailpassword recipe user
+                     * associated with the primary user with the same email
+                     * which is returned by core in the response
+                     */
+                    let epUser = user.loginMethods.find((u) => u.email === email && u.recipeId === "emailpassword");
+                    if (epUser === undefined) {
+                        /**
+                         * create a new emailpassword recipe user
+                         */
+                        let response = await options.recipeImplementation.signUp({
+                            email,
+                            password: newPassword,
+                            userContext,
+                            doAccountLinking: false,
+                        });
+                        if (response.status !== "OK") {
+                            throw Error("this error should not be thrown. EP user already for email: " + email);
+                        }
+                        let recipeUser = response.user;
+                        await verifyUser(response.user.id);
+                        await linkAccounts(recipeUser.id, user.id, userContext);
+                    } else if (!epUser.verified) {
+                        await verifyUser(epUser.recipeUserId);
+                    }
```

Comment:
not as per diagram -> you need to call updatePassword function here + call createPrimaryUserIdOrLinkAccounts?